hplsql/src/main/java/org/apache/hive/hplsql/Exec.java

```diff
@@ -662,6 +660,31 @@ public void registerUdf() {
     udfRegistered = true;
   }
 
+  private static void addJar(ArrayList<String> sql, String dir, String jarName) {
+    if (isNotBlank(jarName)) {
+      sql.add("ADD JAR " + Paths.get(dir, jarName));
+    }
+  }
+
+  private static final Logger LOG = LoggerFactory.getLogger(Exec.class);
+
+  private static String findJarLike(String dir, String prefix) {
+    String[] files = new File(dir).list();
+    if (files == null) {
+      LOG.warn("No jar file found in directory '{}'", dir);
+      return null;
+    }
+
+    for(String jarName: files) {
+      if(jarName.startsWith(prefix) && jarName.endsWith(".jar")) {
+        return jarName;
+      }
+    }
+
+    LOG.warn("No jar file found in directory '{}' with prefix '{}'", dir, prefix);
+    return null;
+  }
```

Comment:
You should use `File.list(FilenameFilter filter)` instead

Reply:
done